src/DotVVM.TypeScript.Compiler/Translators/Operations/OperationTranslatingVisitor.cs

```diff
@@ -0,0 +1,703 @@
+                var literal = _factory.CreateLiteralExpression("1", parent);
+                var binaryOperation = _factory.CreateBinaryOperation(target, @operator, literal, parent);
+                return _factory.CreateAssignment(target as IReferenceSyntax, binaryOperation, parent);
+            }
+            var isIncrement = operation.Kind == OperationKind.Increment;
+            return _factory.CreateIncrementOrDecrement(target, operation.IsPostfix, isIncrement, parent);
+        }
+
+        public override ISyntaxNode VisitForLoop(IForLoopOperation operation, ISyntaxNode parent)
+        {
+            _logger.LogDebug("Operations", "Translating for loop operation.");
+            var beforeStatement = operation.Before.FirstOrDefault().Accept(this, parent) as IStatementSyntax;
+            var condition = operation.Condition.Accept(this, parent) as IExpressionSyntax;
+            var afterStatement = operation.AtLoopBottom.First().Accept(this, parent) as IStatementSyntax;
+            var body = operation.Body.Accept(this, parent) as TsStatementSyntax;
+            return _factory.CreateForStatement(beforeStatement, condition, afterStatement, body, parent);
+        }
+
+        public override ISyntaxNode VisitWhileLoop(IWhileLoopOperation operation, ISyntaxNode parent)
+        {
+            _logger.LogDebug("Operations", "Translating while loop operation.");
+            var condition = operation.Condition.Accept(this, parent) as IExpressionSyntax;
+            var body = operation.Body.Accept(this, parent) as IStatementSyntax;
+            if (operation.ConditionIsTop)
+                return _factory.CreateWhileStatement(condition, body, parent);
+            else
+                return _factory.CreateDoWhileStatement(condition, body, parent);
+        }
+
+        public override ISyntaxNode VisitConditional(IConditionalOperation operation, ISyntaxNode parent)
+        {
+            _logger.LogDebug("Operations", "Translating conditional operation.");
+            var expression = operation.Condition.Accept(this, parent) as IExpressionSyntax;
+            var trueStatement = operation.WhenTrue.Accept(this, parent);
+            var falseStatement = operation.WhenFalse?.Accept(this, parent);
+
+            if (operation.Type == null)
+                return _factory.CreateIfStatement(expression, trueStatement as IStatementSyntax,
+                    falseStatement as IStatementSyntax, parent);
+            else
+                return _factory.CreateConditionalExpression(expression, trueStatement as IExpressionSyntax, falseStatement as IExpressionSyntax, parent);
+        }
+
+        public override ISyntaxNode VisitLiteral(ILiteralOperation operation, ISyntaxNode parent)
+        {
+            var value = "";
+            if (operation.ConstantValue.HasValue) value = operation.ConstantValue.ToString();
+            if (operation.Type.IsEquivalentTo(typeof(string)))
+            {
+                value = $"'{value}'";
```

Comment:
Hey, the string may contain an apostrophe. You'd rather just JSON serialize the value.